### ISSUE TITLE
Can not pass generic function to generic function

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#1208]
The following example …
```py
from typing import Callable

def invoke[A, B](fn: Callable[[A], B], value: A) -> B:
    return fn(value)

def identity[T](x: T) -> T:
    return x

invoke(identity, 1)
```
(https://play.ty.dev/6dfb9951-b78f-4791-af82-151a1c694a7c)

… currently fails with this error:
```
error[invalid-argument-type]: Argument to function `invoke` is incorrect
  --> /home/shark/playground/test.py:12:8
   |
12 | invoke(identity, 1)
   |        ^^^^^^^^ Expected `(Literal[1], /) -> Unknown`, found `def identity[T](x: T@identity) -> T@identity`
   |
info: Function defined here
 --> /home/shark/playground/test.py:4:5
  |
4 | def invoke[A, B](fn: Callable[[A], B], value: A) -> B:
  |     ^^^^^^       -------------------- Parameter declared here
5 |     return fn(value)
  |
info: rule `invalid-argument-type` is enabled by default
```

I am reporting this in addition to #623, because it seems like this is not really related to the shortcomings of the current solver, but rather to the way we handle assignability of the generic `identity` function argument to the call(?).

**Comment by carljm** [comment]
Is this the same as https://github.com/astral-sh/ty/issues/95 ?

**Comment by sharkdp** [comment]
Yes, very much looks like it. I'll copy over my example to make sure it gets resolved when that issue is fixed.

**Comment by sharkdp** [comment]
This is fixable independently from #95, so I'm reopening it for now.
